### PR TITLE
Avoid duplication in package assembly list

### PR DESCRIPTION
**Pull request by mmisol**
### Purpose

Fixes the issue described here https://forum.dynamobim.com/t/package-contents-listed-twice-in-the-package-manager/48994. Tracked as DYN-2602.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@aparajit-pratap @QilongTang

### FYIs

@mjkkirschner

### CODE REVIEW

File: src/DynamoPackages/PackageLoader.cs [hunk 1]
```diff
@@ -320,9 +320,11 @@ public void LoadPackages(IEnumerable<Package> packages)
             // Setting back the DisableRun property back to false, as the package loading is completed.
             EngineController.DisableRun = false;
 
-            var assemblies =
-                enumerable.SelectMany(x => x.EnumerateAssembliesInBinDirectory().Where(y => y.IsNodeLibrary));
-            OnPackagesLoaded(assemblies.Select(x => x.Assembly));
+            var assemblies = packageList
+                .SelectMany(p => p.LoadedAssemblies.Where(y => y.IsNodeLibrary))
+                .Select(a => a.Assembly)
+                .ToList();
```
Comment (mmisol) [inline src/DynamoPackages/PackageLoader.cs:326]: There are basically two things that were fixed:
1. The call to `EnumerateAssembliesInBinDirectory` was replaced by using `LoadedAssemblies`. This avoids re-adding assemblies to `LoadedAssemblies`.
2. The assemblies list is materialized. Before these changes, the call to `EnumerateAssembliesInBinDirectory` was made each time the enumerable was enumerated, which happened once per event handler for `PackageLoaded`. That caused triplication of the libraries for me on latest master.

**Comment by aparajit-pratap** [comment]
@mmisol could youJIRA issue for this for tracking purposes.

File: src/DynamoPackages/PackageLoader.cs (hunk 1 above)
Comment (aparajit-pratap) [reply src/DynamoPackages/PackageLoader.cs:326]: @mmisol you mentioned that `EnumerateAssembliesInBinDirectory` was being called in a couple of other places. Do those need to be replaced with `LoadedAssmeblies` as well?

**Comment by aparajit-pratap** [comment]
@mmisol thanks for fixing this issue so quickly! Are there any tests you can add for this?

**Comment by mjkkirschner** [comment]
should the use of this enumerateAssemblies method also be changed here:
https://github.com/DynamoDS/Dynamo/pull/10550/files#diff-e5fc16a8c62e48d288f3dd8cb3140d9cR261

File: src/DynamoPackages/PackageLoader.cs (hunk 1 above)
Comment (mmisol) [reply src/DynamoPackages/PackageLoader.cs:326]: Yes, you are right. What I was seeing in the stack trace was actually this line of code being executed when the enumeration was materialized. This happened in two event handlers of `OnPackagesLoaded`, one in DynamoModel.cs(978) and another in EngineController.cs(521).

**Comment by mmisol** [comment]
@mjkkirschner That code seems to have the same problem. I'll check if it's being used somewhere.

File: src/DynamoPackages/PackageLoader.cs (hunk 1 above)
Comment (QilongTang) [reply src/DynamoPackages/PackageLoader.cs:326]: @mmisol So I assume whichever event handler called second time, with this fix, we will no longer re-adding assemblies to `LoadedAssemblies`. Do we want to create a test by mocking up the situation and check the end `LoadedAssemblies`?

Comment (mmisol) [reply src/DynamoPackages/PackageLoader.cs:326]: Yes, that was part of the issue. The call to `TryLoadPackageIntoLibrary` is the first to call `EnumerateAssembliesInBinDirectory` and after this `LoadedAssemblies` should be used to read them. 

I'll see to add a test. I have to think about what needs to be mocked, hopefully it's not too difficult.

**Comment by mmisol** [comment]
Added a test. This is ready for review @aparajit-pratap @QilongTang 

@mjkkirschner About the method you mentioned, it's obsoleted and there are no usages, so I left it alone. Let me know if you would like to change ti anyway, but testing it would be difficult.

**Comment by QilongTang** [comment]
@mmisol Thank you for adding the test. LGTM. Not sure if we need to include the other spot also part of this PR?

**Comment by mmisol** [comment]
Re-thinking about what I wrote yesterday, unit testing the obsolete method should be easy. If you guys think it's safer to change it and unit test it rather than leaving as-is let me know and I can change it.

**Comment by aparajit-pratap** [comment]
> Re-thinking about what I wrote yesterday, unit testing the obsolete method should be easy. If you guys think it's safer to change it and unit test it rather than leaving as-is let me know and I can change it.

I think this is good. LGTM.

**Comment by mmisol** [comment]
Is this PR good to be merged @QilongTang @aparajit-pratap @mjkkirschner ? If so, can I get an approval? Thank you

**Comment by QilongTang** [comment]
@mmisol Ship it!

**Pull request merged**